airsonic-main/src/test/java/org/airsonic/player/dao/AlbumDaoTest.java

```diff
@@ -0,0 +1,155 @@
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class AlbumDaoTest extends AbstractAirsonicHomeTest {
+
+    private List<MusicFolder> musicFolders;
+
+    @Autowired
+    private MediaFileDao mediaFileDao;
+
+    @Autowired
+    private AlbumDao albumDao;
+
+    private List<MusicFolder> sameName;
+    private List<MusicFolder> noTagFirst;
+    private List<MusicFolder> nameInReverse;
+
+    private final int offset = 0;
+    private final int count = Integer.MAX_VALUE;
+
+    @Before
+    public void setup() {
+        if (ObjectUtils.isEmpty(musicFolders)) {
+            sameName = Arrays.asList(new MusicFolder(1, new File(resolveBaseMediaPath.apply("Scan/ID3Album/sameName")), "1", true, new Date()));
+            noTagFirst = Arrays.asList(new MusicFolder(2, new File(resolveBaseMediaPath.apply("Scan/ID3Album/noTagFirst")), "2", true, new Date()));
+            nameInReverse = Arrays.asList(new MusicFolder(3, new File(resolveBaseMediaPath.apply("Scan/ID3Album/nameInReverse")), "3", true, new Date()));
+            musicFolders = new ArrayList<>();
+            musicFolders.addAll(sameName);
+            musicFolders.addAll(noTagFirst);
+            musicFolders.addAll(nameInReverse);
+        }
+        populateDatabaseOnlyOnce();
+    }
+
+    @Override
+    public List<MusicFolder> getMusicFolders() {
+        return musicFolders;
+    }
+
+    @Test
+    public void testUpdateAlbum() {
+
+        // File structure
+        List<MediaFile> albums = mediaFileDao.getAlphabeticalAlbums(offset, count, false, sameName);
+        assertEquals(1, albums.size());
+        MediaFile album = albums.get(0);
+        assertEquals("ALBUM1", album.getName());
+        assertEquals("albumArtistA", album.getArtist());
+        assertNull(album.getAlbumArtist());
+        assertEquals("genreA", album.getGenre());
+        assertEquals(Integer.valueOf(2001), album.getYear());
+
+        // ID3 tags
+        List<Album> albumId3s = albumDao.getAlphabeticalAlbums(offset, count, false, false, sameName);
+        assertEquals(1, albumId3s.size());
+        Album albumId3 = albumId3s.get(0);
+
+        // [albumA albumArtistA genreB 2002]
+        // This combination is "a combination that should not exist.
+        assertEquals("albumA", albumId3.getName());
+        assertEquals("albumArtistA", albumId3.getArtist());
+        assertEquals("genreB", albumId3.getGenre()); // XXX Unintentional overwrite
+        assertEquals(Integer.valueOf(2002), albumId3.getYear()); // XXX Unintentional overwrite
+
+        assertNull(album.getMusicBrainzReleaseId());
+        assertNull(album.getMusicBrainzRecordingId());
+    }
+
+    @Test
+    public void testNoTagFirst() {
+
+        // File structure
+        List<MediaFile> albums = mediaFileDao.getAlphabeticalAlbums(offset, count, false, noTagFirst);
+        assertEquals(1, albums.size());
+        MediaFile album = albums.get(0);
```

Comment:
Instead of using `get(0)`, how about selecting files by path? The order is not important here (and might be different in different platforms).

Reply:
Uh.
I can't immediately think of the cause of this. As you say, let's rewrite because the latter case of assertNull is the subject.

Reply:
I have tracked down this issue and the cause is due to an HSQLDB update. I created this test before updating HSQLDB. Results will change before and after updating HSQLDB.

> and might be different in different platforms

With POSIX, at least the range of the alphabet seems to be the same ... I wonder a little :confused:
Methods such as getAlphabeticalAlbums and getChildrenOf that implicitly follow the record registration order may return results in a different order than before. Of course, I believe that all business logic is properly sorted after that. If not, it will have an impact.

As you pointed out, there is no direct relationship between this test case and its problem.

Reply:
A little discussion in me ...

It has already been confirmed that some methods that do not guarantee the order of the results will change the order by the update of HSQLDB. 
But is this case valid ...